pushtest/test_webpush.py

```diff
@@ -0,0 +1,162 @@
+    client = quick_register(url, use_webpush=True)
+    client.disconnect()
+    ok_(client.channels)
+    client.send_notification(data=data, status=202)
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    eq_(result["data"], data)
+
+    client.disconnect()
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    eq_(result["data"], data)
+
+
+def test_multiple_delivery_repeat_without_ack(url=None):
+    data = str(uuid.uuid4())
+    data2 = str(uuid.uuid4())
+    url = url or check_environ()
+    client = quick_register(url, use_webpush=True)
+    client.disconnect()
+    ok_(client.channels)
+    client.send_notification(data=data, status=202)
+    client.send_notification(data=data2, status=202)
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+
+    client.disconnect()
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+
+
+def test_multiple_delivery_with_single_ack(url=None):
+    data = str(uuid.uuid4())
+    data2 = str(uuid.uuid4())
+    url = url or check_environ()
+    client = quick_register(url, use_webpush=True)
+    client.disconnect()
+    ok_(client.channels)
+    client.send_notification(data=data, status=202)
+    client.send_notification(data=data2, status=202)
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    client.ack(result["channelID"], result["version"])
+
+    client.disconnect()
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    ok_(result["messageType"], "notification")
+    result = client.get_notification()
+    eq_(result, None)
+
+
+def test_multiple_delivery_with_multiple_ack(url=None):
+    data = str(uuid.uuid4())
+    data2 = str(uuid.uuid4())
+    url = url or check_environ()
+    client = quick_register(url, use_webpush=True)
+    client.disconnect()
+    ok_(client.channels)
+    client.send_notification(data=data, status=202)
+    client.send_notification(data=data2, status=202)
+    client.connect()
+    client.hello()
+    result = client.get_notification()
+    ok_(result != {})
+    ok_(result["data"] in [data, data2])
+    result2 = client.get_notification()
+    ok_(result2 != {})
+    ok_(result2["data"] in [data, data2])
+    client.ack(result2["channelID"], result2["version"])
```

Comment:
Well, technically, version is any incrementing value. If one isn't provided, we default to using a timestamp. I had originally used milliseconds in order to limit how many messages we could get hit with at a given time. This was when data wasn't included and we would effectively collapse events.